File: backend/semgrep_rules/tests/javascript/kisa_sw_14_integer_overflow.js
function unsafe(req) {
  const value = req.query.value;
  // ruleid: kisa.sw14.javascript.external-input-to-bitwise-int
  return value | 0;
}
function safe(req) {
  const value = Number(req.query.value);
  if (!Number.isSafeInteger(value)) throw new Error("invalid");
  // ok: kisa.sw14.javascript.external-input-to-bitwise-int
  return value;
}

File: backend/semgrep_rules/tests/java/kisa_sw_14_integer_overflow.java
import javax.servlet.http.HttpServletRequest;
class kisa_sw_14_integer_overflow {
    int unsafe(HttpServletRequest req, int count) {
        int size = Integer.parseInt(req.getParameter("size"));
        // ruleid: kisa.sw14.java.external-input-arithmetic-review
        return size * count;
    }
    long safe(HttpServletRequest req, int count) {
        long size = Long.parseLong(req.getParameter("size"));
        // ok: kisa.sw14.java.external-input-arithmetic-review
        return Math.multiplyExact(size, count);
    }
}
